idea/src/org/jetbrains/kotlin/idea/intentions/JoinDeclarationAndAssignmentIntention.kt

```diff
@@ -56,12 +57,14 @@ class JoinDeclarationAndAssignmentIntention : SelfTargetingRangeIntention<KtProp
         }
 
         val assignment = findAssignment(element) ?: return null
-        if (assignment.right?.let {
+        val assignmentRight = assignment.right ?: return null
```

Comment:
Please call `analyze()` once and use the resulting `BindingContext` instead of `mainReference.resolve()`. The rest is OK 👍